src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -351,7 +351,7 @@ CATCH_RETURN()
             _glyphIndices.resize(totalGlyphsArrayCount);
         }
 
-        if (_isEntireTextSimple)
+        if (_isEntireTextSimple && !_fontRenderData->DidUserSetFeatures())

```

Comment:
I am a bi confused by this. Say without this PR, we have ligature enabled by default, and we think some text can be simple. But with this PR and people disabling ligature, suddenly all text is complicated now? If anything, isn’t  text w/o ligature a “simple” one instead?

I mean code-wise, I get it. You have to pass the Feature structure in. But it seems wrong logically.

Reply:
The thing is we cannot detect what glyphs get affected/don't get affected based on how the user changed the default features. So as long as the user changed something, we cannot take the shortcut and need to call `GetGlyphs` instead